src/record_syscall.cc

```diff
@@ -6006,6 +6006,44 @@ static vector<WriteHole> find_holes(RecordTask* t, int desc, uint64_t offset, ui
   return ret;
 }
 
+static void check_outside_mappings(const string& tracee_file_name, const RecordSession& session) {
+  DIR* proc = opendir("/proc");
+  if (!proc) {
+    FATAL() << "Failed to read /proc";
+  }
+  while (true) {
+    struct dirent* f = readdir(proc);
+    if (!f) {
+      break;
+    }
+    pid_t pid = atoi(f->d_name);
+    if (pid) {
+      bool ok = true;
+      for (auto it : session.tasks()) {
+        if (pid == it.second->tid) {
```

Comment:
Why don't you use `Session::find_task` instead of this loop?

Reply:
Thanks for looking into it. I was not aware of it. That makes it looking much nicer.